OSDOCS-5890 Zstream release notes for 4.11.39 (#59329)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

[id="ocp-4-11-38-upgrading"]
==== Updating

To update an existing {product-title} 4.11 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster within a minor version by using the CLI] for instructions.

[id="ocp-4-11-39"]
=== RHSA-2023:2014 - {product-title} 4.11.39 bug fix and security update

Issued: 2023-05-02

> {product-title} release 4.11.39 is now available. Bug fixes included in the update are listed in the link:https://access.redhat.com/errata/RHSA-2023:2014[RHSA-2023:2014] advisory. RPM packages included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:2013[RHBA-2023:2013] advisory.

Author: Done

> {product-title} release 4.11.39, which includes security updates, is now available. Bug fixes included in the update are listed in the link:https://access.redhat.com/errata/RHSA-2023:2014[RHSA-2023:2014] advisory. RPM packages included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:2013[RHBA-2023:2013] advisory.

Author: Changed, thanks for the catch!

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.11.39 --pullspecs
----

[id="ocp-4-11-39-bug-fixes"]
==== Bug fixes

> * Previously, the Ingress Operator failed to publish `router-certs` secret on a cluster with a high number of Ingress Controllers because of the secret size limit of 1 mebibyte. Consequently, the Authentication Operator that uses the `router-certs` secret to access the cluster Ingress domain might not have had the latest certificate to use for OAuth purposes. With this update, the Ingress Operator can generate the default certificate and key for each Ingress Controller, so that the Ingress Operator can publish a `router-certs` secret on the cluster. This update ensures that the Authentication Operator can read and use an up-to-date certificate for OAuth authentication. (link:https://issues.redhat.com/browse/OCPBUGS-8000[*OCPBUGS-8000*])

Reviewer: * There is a specific secret named "router-certs", so it makes sense to write "_the_ `router-certs` secret".
* There is only one Authentication Operator, and it uses the `router-certs` secret, so I think it is appropriate to write "the Authentication Operator, _which_ uses the `router-certs` secret"¸ using the non-restrictive "which" rather than the restrictive "that".
* The Ingress Operator does generate a certificate and key for each Ingress Controller, but that part was working fine before the bug fix, so it seems confusing to me to say, "With this update, the Ingress Operator can generate" it.  The bug fix is that now the Ingress Operator can _publish_ the `router-certs` secret even if there is a high number of Ingress Controllers.  
* The key change is that the Ingress Operator _no longer_ publishes the certificate and key for each Ingress Controller; the Ingress Operator now only publishes the certificate and key that the Authentication Operator needs, namely the certificate and key for the cluster Ingress domain.

```suggestion
* Previously, the Ingress Operator failed to publish the `router-certs` secret on a cluster with a high number of Ingress Controllers because of the secret size limit of 1 mebibyte. Consequently, the Authentication Operator, which uses the `router-certs` secret to access the cluster Ingress domain, might not have had the latest certificate to use for OAuth purposes. With this update, the Ingress Operator publishes the certificate and key only for the Ingress Controller that owns the cluster Ingress domain, so that the secret doesn't exceed the size limit. This update ensures that the Authentication Operator can read and use an up-to-date certificate for OAuth authentication. (link:https://issues.redhat.com/browse/OCPBUGS-8000[*OCPBUGS-8000*])
```

Author: Added changes.

> * Previously, the Ingress Operator failed to publish the `router-certs` secret on a cluster with a high number of Ingress Controllers because of the secret size limit of 1 mebibyte. Consequently, the Authentication Operator, which uses the `router-certs` secret to access the cluster Ingress domain, might not have had the latest certificate to use for OAuth purposes. With this update, the Ingress Operator publishes the certificate and key only for the Ingress Controller that owns the cluster Ingress domain, so that the secret doesn't exceed the size limit. This update ensures that the Authentication Operator can read and use an up-to-date certificate for OAuth authentication. (link:https://issues.redhat.com/browse/OCPBUGS-8000[*OCPBUGS-8000*])

Reviewer 2: ```suggestion
* Before this update, the Ingress Operator failed to publish the `router-certs` secret on a cluster with a high number of Ingress Controllers because of the secret size limit of 1 MB. Consequently, the Authentication Operator, which uses the `router-certs` secret to access the cluster Ingress domain, might not have had the latest certificate to use for OAuth purposes. With this update, the Ingress Operator publishes the certificate and key only for the Ingress Controller that owns the cluster Ingress domain, so that the secret does not exceed the size limit. This update ensures that the Authentication Operator can read and use an up-to-date certificate for OAuth authentication. (link:https://issues.redhat.com/browse/OCPBUGS-8000[*OCPBUGS-8000*])
```
For release notes, a suggestion is to use 'Before this update' rather than 'Previously'.
See the ISG (contractions): https://www.ibm.com/docs/en/ibm-style?topic=grammar-contractions
See the ISG (units of measure): https://www.ibm.com/docs/en/ibm-style?topic=measurement-units

Author: Thanks!

Author: I talked about it with Joe and Darragh and we think Previously is okay, for consistency.

> * Previously, the Ingress Operator published certificates and keys of all Ingress Controllers to the `router-certs` secret in the `openshift-config-managed` namespace. The Authentication Operator is the only operator that uses the `router-certs` secret to access the cluster Ingress domain. Publishing unnecessary certificates and keys can impact the generation of the secret, because the secret exceeds the maximum resource size limit. For the {product-title} {product-version} release, the Ingress Operator publishes certificates and keys for any Ingress Controllers that relate to the cluster Ingress domain. This change ensures that the Ingress Operator can generate a `router-certs` secret that the Authentication Operator can access for OAuth authentication purposes.  (link:https://issues.redhat.com/browse/OCPBUGS-8000[*OCPBUGS-8000*])

Reviewer 3: ```suggestion
* Previously, the Ingress Operator failed to publish the `router-certs` secret on a cluster with many Ingress Controllers  because of the secret size limit of 1 mebibyte. Consequently, the Authentication Operator that uses the `router-certs` secret to access the cluster Ingress domain might not have the latest certificate to use for OAuth purposes. With this update, the Ingress Operator publishes the default certificate and key for the Ingress Controller, which has the cluster Ingress domain. This update ensures that the Ingress Operator no longer fails to publish the `router-certs` secret and the Authentication Operator can read and use an up-to-date certificate for OAuth authentication. (link:https://issues.redhat.com/browse/OCPBUGS-8000[*OCPBUGS-8000*])
```
